src/django_components/slots.py

```diff
@@ -62,45 +60,52 @@ class SlotFunc(Protocol, Generic[TSlotData]):
     def __call__(self, ctx: Context, slot_data: TSlotData, slot_ref: "SlotRef") -> SlotResult: ...  # noqa E704
 
 
-SlotContent = Union[SlotResult, SlotFunc[TSlotData]]
+SlotContent = Union[SlotResult, SlotFunc[TSlotData], "Slot[TSlotData]"]
 
-# Internal type aliases
-SlotId = str
-SlotName = str
 
+@dataclass
+class Slot(Generic[TSlotData]):
+    """This class holds the slot content function along with related metadata."""
 
-class Slot(NamedTuple):
```

Comment:
NOTE: The old and new `Slot` classes share only the name, but they are used for different purposes.